### PR TITLE
feat: create dashboard

### PR DESCRIPTION
**Pull request by tracy-french**
# Description

This change includes the complete create dashboard form. Currently, the form is only accessible directly by URL. A link to the page will be made available in the upcoming dashboards page.

The change also includes updates to the dashboard entity:
- Max length dashboard name is now 40 characters.
- Max length dashboard description is now 200 characters.
- Min length dashboard description is now 1 character.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Build passing locally.
- [x] e2e test suite passing locally.

Screenshot 2023-03-22 at 14 23 14

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: apps/client/src/routes/dashboards/create/hooks/use-create-dashboard-mutation.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          <Button onClick={() => navigate(`/dashboards/${newDashboard.id}`)}>
+            <FormattedMessage
+              defaultMessage="View dashboard"
+              description="create dashboard success notification action"
+            />
+          </Button>
+        ),
+      });
+
+      navigate(DASHBOARDS_HREF);
```
Comment (chejimmy) [inline apps/client/src/routes/dashboards/create/hooks/use-create-dashboard-mutation.tsx:53]: nit: want this to fire only if we are still on the create flow. We can put the `navigate(...)` in the component so it fires only if the Ui still showing [example](https://tkdodo.eu/blog/mastering-mutations-in-react-query#some-callbacks-might-not-fire)

**Review by chejimmy — APPROVED** [review]

File: apps/client/src/routes/dashboards/create/create-dashboard-route.tsx [hunk 2]
```diff
@@ -20,20 +19,3 @@ export const route = {
     format: CREATE_DASHBOARD_PAGE_FORMAT,
   },
 } satisfies Parameters<typeof createBrowserRouter>[0][number];
```
[the remark this answers is not in the document]
Comment (tracy-french) [reply apps/client/src/routes/dashboards/create/create-dashboard-route.tsx:21]: It's doesn't seem to work lol. There's also `DataRouteObject` and `IndexRouteObject`,

File: apps/core/src/dashboards/entities/dashboard.entity.ts [hunk 3]
```diff
@@ -19,14 +19,14 @@ export class Dashboard {
    * @example "Wind Farm 4"
    */
   @IsString()
-  @Length(1, 100)
+  @Length(1, 40)
```
[the remark this answers is not in the document]
Comment (tracy-french) [reply apps/core/src/dashboards/entities/dashboard.entity.ts:22]: 40 is the max length of Cloudscape text input before scrolling. :D

File: apps/client/src/routes/dashboards/create/hooks/use-create-dashboard-mutation.tsx (hunk 1 above)
Comment (tracy-french) [reply apps/client/src/routes/dashboards/create/hooks/use-create-dashboard-mutation.tsx:53]: Good catch! Also, this is a fantastic blog for react-query!

**Pull request merged**